### PR TITLE
AMBARI-24731 - Infra Manager: scheduled cleanup of old jobs

### PR DESCRIPTION
**Pull request by kasakrisz**
## What changes were proposed in this pull request?

- Create clean up job for deleting expired job data from Infra Manager local database
- Replace aws sdk to minio because aws sdk can not be modularized and it was not worked with localstack which we use for mock s3 in IT
- Refactor job configuration: prior the refactor only job groups could be defined from now single jobs too
- Extract Solr specific functionality from JobScheduler to get a more general scheduler. 
- Use immutable configuration and parameter objects for jobs
- Adjust integration test for the separate infra repo

## How was this patch tested?

1. run Integration test 
2. manually:
- Deploy Ambari and a cluster with infra solr, infra manager, logsearch, hdfs
- enable archive_service_job set the destination to hdfs and run it using infra manager API
- check hdfs contains files in the archives destination directory. Check the content of the files
3. manually:
- start infra manager in a docker environment
- set cleanup schedule to run the job in every minute and set the ttl for a small duration of time
- check infra managers sqlite db contains job data and their creation date

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-Infra-Github-PR-Builder/4/
Test PASSed.

File: ambari-infra-manager/src/main/java/org/apache/ambari/infra/job/archive/S3Uploader.java [hunk 1]
```diff
@@ -48,14 +64,29 @@ public S3Uploader(S3Properties s3Properties, PasswordStore passwordStore) {
...
+      if (!client.bucketExists(bucketName))
+        client.makeBucket(bucketName);
+
+    } catch (RegionConflictException | XmlPullParserException | InvalidBucketNameException | NoSuchAlgorithmException | InsufficientDataException | ErrorResponseException | InvalidKeyException | NoResponseException | InvalidPortException | InvalidEndpointException | InternalException e) {
+      throw new RuntimeException(e);
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+
+//    BasicAWSCredentials credentials = new BasicAWSCredentials(
```
Comment (oleewere) [inline ambari-infra-manager/src/main/java/org/apache/ambari/infra/job/archive/S3Uploader.java]: remove this

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-Infra-Github-PR-Builder/5/
Test PASSed.

**Comment by kasakrisz** [comment]
@oleewere I removed the commented parts.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-Infra-Github-PR-Builder/8/
Test PASSed.

**Pull request merged**